x/executionlayer/handler.go

```diff
@@ -68,45 +68,78 @@ func handlerMsgCreateValidator(ctx sdk.Context, k ExecutionLayerKeeper, msg type
 
 	k.SetValidator(ctx, eeAddress, validator)
 
-	return getResult(true, msg)
+	return getResult(true, "")
 }
 
 func handlerMsgBond(ctx sdk.Context, k ExecutionLayerKeeper, msg types.MsgBond) sdk.Result {
-	err := k.Execute(ctx, msg.FromPubkey, msg.TokenContractAddress, msg.SessionCode, msg.SessionArgs, msg.PaymentCode, msg.PaymentArgs, msg.GasPrice)
-	if err != nil {
-		return getResult(false, msg)
-	}
-	return getResult(true, msg)
+	result, log := execute(ctx, k, msg.MsgExecute)
+	return getResult(result, log)
 }
 
 func handlerMsgUnBond(ctx sdk.Context, k ExecutionLayerKeeper, msg types.MsgUnBond) sdk.Result {
-	err := k.Execute(ctx, msg.FromPubkey, msg.TokenContractAddress, msg.SessionCode, msg.SessionArgs, msg.PaymentCode, msg.PaymentArgs, msg.GasPrice)
+	result, log := execute(ctx, k, msg.MsgExecute)
+
+	return getResult(result, log)
+}
+
+func execute(ctx sdk.Context, k ExecutionLayerKeeper, msg types.MsgExecute) (bool, string) {
+
+	// Parameter preparation
+	stateHash := ctx.CandidateBlock().State
+	protocolVersion := k.MustGetProtocolVersion(ctx)
+	exexAddr := sdk.GetEEAddressFromSecp256k1PubKey(msg.ExecPubkey)
+	log := ""
+
+	// Execute
+	deploys := []*ipc.DeployItem{}
```

Comment:
if reason require to deploys(slice) is for ExecuteRequest parameter,
you can move deploy to slice initialize without append.
I think so it's saving a little copy cost.

ex)
deploys := []&ipc.DeployItem{
 util.MakeDeploy(...),
}

Reply:
https://github.com/hdac-io/friday/pull/60/commits/514317830ca571db6be2cdd0a80ac56bb3f2e72d